hastory-server/src/Data/Hastory/Server/Handler/Users.hs

```diff
@@ -2,14 +2,15 @@
 
 module Data.Hastory.Server.Handler.Users where
 
-import Data.Hastory.Server.Handler.Import
+import           Data.Hastory.Server.Handler.Import
 
 createUserHandler :: ServerSettings -> UserForm -> Handler UserId
-createUserHandler ServerSettings {..} UserForm {..} = do
+createUserHandler ServerSettings {..} userForm@UserForm {..} = do
   mUser <- runDB (getBy $ UniqueUsername userFormUserName) _ssDbPool
-  case mUser of
-    Just _ -> throwError err400
-    Nothing -> buildAndInsertUser userFormUserName userFormPassword
+  let validForm = isNothing mUser || isValid userForm
+  if validForm
+    then throwError err400
```

Comment:
throw "bad request" if the form is valid?
and what does valid mean here? there is no user yet OR the userForm is valid?
What's the reasoning here?

Reply:
OK - so when I wrote this code, I was either sleep coding, a git command went wrong. Something wasn't right. I've now fixed the logic, but I think in a separate comment you mentioned that things can be improved, so I am going to mark this as resolved.